testsuite/MDAnalysisTests/topology/test_itp.py

```diff
@@ -335,3 +335,14 @@ def test_missing_endif(self):
         with pytest.raises(IOError):
             with self.parser(ITP_no_endif) as p:
                 top = p.parse(include_dir=GMX_DIR)
+
+class Testrelativepath:
+
+    def test_relpath(self, tmp_path):
+        content="[ atoms ]\n" +\
+                "     1      H      1    SOL    HW1      1       0.41    1.00800"
+        d = tmp_path / "sub"
+        d.mkdir()
+        p = d / "test.itp"
+        p.write_text(content)
+        u = mda.Universe(tmp_path / "sub/test.itp", format='ITP')
```

Comment:
I think this would still be an absolute path. Instead, I would suggest using the `tmpdir` fixture, creating a new directory inside that temporary one, and making sure that the relative path you test looks something like "../test.itp".

Reply:
Hi @lilyminium, I've been looking at pytest tools and os.paths. I see that there's a command called relative_to in the pathlib module which computes a relative path to a specified directory. My approach would have been to make a tmpdir and set that as 'observer' directory and use that command (relative_to) to find the path that it sees. 

As it turns out, the relative_to command only works when computing paths from parent directory reference, not sibling directory. So, I can't compute something like '../dir'. The second thing is even if I manually create a relative string which tmpdir would use  to read my temporary itp file, I am not sure how to ask pytest to use my tmpdir while performing this test. 

I might have missed something here, especially because I am new to pytest, but I am happy to hear any advice you have here!

Reply:
Hi @aditya-kamath,

If you go into your temporary directory, write a file called "test.itp", make a new subdirectory, and change into that directory, you should be able to test that file with the path "../test.itp". You might like the tmpdir.as_cwd context manager for that.

Reply:
Great advice! I figured the tests out. Just a note that I've done the reverse of what you might expect. Instead of testing a relative path and then converting it to a relative string. I've first tested a relative string and then followed it up by converting it to a path. Hope that's alright!